src/utils.rs

```diff
@@ -268,15 +268,37 @@ pub(crate) fn now_systime(clock: &dyn Clock) -> SystemTime {
         .expect("Failed to convert Clock time to SystemTime")
 }
 
+/// Computes the "index key" (lowest bound) for an SST index block, ie a key that's greater
+/// than all keys in the previous block and less than or equal to all keys in the new block
+pub(crate) fn index_key(prev_block_last_key: Option<Bytes>, this_block_first_key: &Bytes) -> Bytes {
+    if prev_block_last_key.is_none() {
+        Bytes::new()
+    } else {
+        compute_lower_bound(&prev_block_last_key.unwrap(), this_block_first_key)
+    }
+}
+
+fn compute_lower_bound(prev_block_last_key: &Bytes, this_block_first_key: &Bytes) -> Bytes {
+    for i in 0..prev_block_last_key.len() {
+        if prev_block_last_key[i] != this_block_first_key[i] {
+            return Bytes::copy_from_slice(&this_block_first_key[..i + 1]);
+        }
+    }
+
+    // if we didn't find a mismatch yet then the prev block's key must be shorter,
+    // so just use the common prefix plus the next byte in this block's key
+    Bytes::copy_from_slice(&this_block_first_key[..prev_block_last_key.len() + 1])
```

Comment:
is there a way to do this without copying?

Reply:
```suggestion
    this_block_first_key.slice(..prev_block_last_key.len() + 1)
```